docs/GooglePublisher-batchsink.md

```diff
@@ -0,0 +1,46 @@
+# Google Cloud PubSub Sink
+
+Description
+-----------
+This sink writes to a Google Cloud Pub/Sub topic.
+Cloud Pub/Sub brings the scalability, flexibility, and reliability of enterprise message-oriented
+middleware to the cloud. By providing many-to-many, asynchronous messaging that decouples senders and receivers,
+it allows for secure and highly available communication between independently written applications.
+
+Authorization
+-------------
+If the plugin is run on a Google Cloud Dataproc cluster, the service account key does not need to be provided.
+Credentials will be automatically read from the cluster environment.
+
+If the plugin is not run on a Dataproc cluster, the path to a service account key must be provided.
+The service account key can be found on the Dashboard in the Cloud Platform Console.
+Make sure the account key has permission to access Google Cloud Pub/Sub.
+The service account key file needs to be available on every node in your cluster and
+must be readable by all users running the job.
+
+Properties
+----------
+**Project ID**: The Google Cloud Project ID, which uniquely identifies a project.
+It can be found on the Dashboard in the Google Cloud Platform Console.
+
+**Service Account File Path**: Path on the local file system of the service account key used for
+authorization. Does not need to be specified when running on a Dataproc cluster.
+When running on other clusters, the file must be present on every node in the cluster.
+
+**Topic**: Name of the Google Cloud PubSub topic to publish to.
+
+**MessageCountBatchSize**: Pub/Sub Publisher supports batching of messages and you can use this field to configure
+the maximum count of messages in a batch. The default value is 100.
+
+**RequestThresholdKB**: Pub/Sub Publisher supports batching of messages and you can use this field to configure
+the maximum size of a batch in kilo bytes. The default value is 1 KB.
+
+**PublishDelayThresholdMillis**: Pub/Sub Publisher supports batching of messages and you can use this field to configure
+the maximum delay in milli seconds after which the pending messages in the batch will be published.
+The default value is 1 milli second.
+
+**RetryTimeoutSeconds**: Pub/Sub Publisher supports retrying message publish on failures and you can use this field to
+configure the maximum amount of time in seconds to spend retrying publishing failures. The default value is 30 seconds.
+
+**ErrorThreshold**: Maximum number of message publishing
```

Comment:
**Error Threshold**: Maximum number of messages that failed to publish per partition before the pipeline will be failed. The default value is 0.